src/core/HY_Features_MPI.cpp

```diff
@@ -0,0 +1,55 @@
+#include <HY_Features_MPI.hpp>
+#include <HY_PointHydroNexusRemote.hpp>
+
+using namespace hy_features;
+
+HY_Features_MPI::HY_Features_MPI( PartitionData partition_data, geojson::GeoJSON linked_hydro_fabric, std::shared_ptr<Formulation_Manager> formulations, int mpi_rank, int mpi_num_procs) :
+      network(linked_hydro_fabric), formulations(formulations), mpi_rank(mpi_rank), mpi_num_procs(mpi_num_procs)
+{ 
+      std::string feat_id;
+      std::string feat_type;
+      std::vector<std::string> origins, destinations;
+      
+      std::unordered_map<std::string, HY_PointHydroNexusRemote::catcment_location_map_t> remote_connections;
```

Comment:
Since this map has no external dependence, it should probably just be a member of PartitionData